sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs

```diff
@@ -326,12 +327,40 @@ internal static void CopyHeaders(HttpHeaders from, HttpHeaders to)
 #if NET6_0_OR_GREATER
         private static string JoinHeaderValues(HeaderStringValues values)
         {
-            return values.Count switch
-            {
-                0 => string.Empty,
-                1 => values.ToString(),
-                _ => string.Join(",", values)
-            };
+            var count = values.Count;
+            // Explicitly handle most common cases
+            switch (count) {
+                case 1:
```

Comment:
Curious: why are cases not in numerical order?

Reply:
`1` is the most common case. But since IL creates `switch` instruction anyway, it can be ordered in numerical order

Reply:
Replaced with `DefaultInterpolatedStringHandler`.